### PR TITLE
[New] Group layers together

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR implements `Group layers together` in `Layers` category.
URL to README: [ReadME](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-GroupLayersTogether/Shared/Samples/Group%20layers%20together)

## Linked Issue(s)

- `swift/issues/4075`

## How To Test

- Tap the "Layers" button to bring up the layers sheet.
- For project area group layers, toggle the switch next to a layer's name to change its visibility.
- For the buildings group layers, tap the cell with the layer's name to change its visibility.
- Toggle the switch next to a group layer's name to change the visibility of the whole group.

## Screenshots

group-layers-together

### CODE REVIEW

File: Shared/Samples/Group layers together/GroupLayersTogetherView.swift [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                }
+            }
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    Button("Layers") {
+                        isShowingLayersSheet = true
+                    }
+                    .sheet(isPresented: $isShowingLayersSheet, detents: [.medium]) {
+                        layersList
+                    }
```
Comment (des12437) [inline Shared/Samples/Group layers together/GroupLayersTogetherView.swift:62]: The settings sheet covers most of the layers on iPhone. The layer visibility can be fixed by refactoring the button and sheet using [fraction](https://developer.apple.com/documentation/swiftui/presentationdetent/fraction%28_%3A%29) presentation detents. Here is an [example](https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/main/Shared/Samples/Add%20dynamic%20entity%20layer/AddDynamicEntityLayerView.swift#L73-L97) using the fraction presentation detents.

Comment (CalebRas) [reply Shared/Samples/Group layers together/GroupLayersTogetherView.swift]: Thanks for the feedback! I added the fraction presentation, but I am not sure if it is what you had in mind, so let know if it needs more work.

**Review by mhdostal — DISMISSED** [review]

File: Shared/Samples/Group layers together/GroupLayersTogetherView.swift [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+#endif
+                            }
+                    } else {
+                        layersButton
+                            .sheet(isPresented: $isShowingLayersSheet, detents: [.medium]) {
+                                layersList
+                            }
+                    }
+                }
+            }
```
Comment (des12437) [inline Shared/Samples/Group layers together/GroupLayersTogetherView.swift:79]: Thank you for making this change. The logic should be refactored out of the toolbar.
```suggestion
     layersButton
```

```
@ViewBuilder private var layersButton: some View {
        let button = Button("Layers") {
            isShowingLayersSheet = true
        }
        
        if #available(iOS 16, *) {
            button
                .popover(isPresented: $isShowingLayersSheet, arrowEdge: .bottom) {
                    layersList
                        .presentationDetents([.fraction(0.5)])
#if targetEnvironment(macCatalyst)
                        .frame(minWidth: 300, minHeight: 270)
#else
                        .frame(minWidth: 320, minHeight: 390)
#endif
                }
        } else {
            button
                .sheet(isPresented: $isShowingLayersSheet, detents: [.medium]) {
                    layersList
                }
        }
    }
```

**Review by des12437 — APPROVED** [review]

**Review by mhdostal — APPROVED** [review]

File: Samples.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1916,6 +1934,7 @@
 				"CODE_SIGN_IDENTITY[sdk=macosx*]" = "Apple Development";
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 1;
+				DEVELOPMENT_TEAM = "";
```
Comment (yo1995) [inline Samples.xcodeproj/project.pbxproj:1937]: Not a big deal, pay attention to changes like this in the future. Sometimes easy-to-be-ignored minor changes may slip into a PR that causes unexpected side effect.

**Pull request merged**